### PR TITLE
BitBucket Repository Push

### PR DESCRIPTION
**Pull request by PeyGis**
# Description

Added documentation on streaming the latest BitBucket commit hash to Port using webhooks.

## Added docs pages

docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+sidebar_position: 10
+description: Keep track of latest commits in your catalog
+---
+
+import CommitBlueprint from './resources/bitbucket/\_example_bitbucket_push_blueprint.mdx'
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:6]: Let's rename to `MicroserviceBlueprint`

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# BitBucket
+
+In this example you are going to create a webhook integration between [BitBucket](https://bitbucket.org) and Port, which will ingest commit entities.
+
+## Prerequisites
+
+Create the following blueprint definition and webhook configuration:
+
+<details>
+<summary>Commit blueprint</summary>
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:18]: Here as well

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+<CommitBlueprint/>
+
+</details>
+
+<details>
+<summary>Commit webhook configuration</summary>
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:25]: Maybe Push webhook configuration instead

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+## Create the BitBucket webhook
+
+1. Go to your desired repository in BitBucket;
+2. Select **Repository Settings**;
+3. Select **Webhooks**;
+4. Click on **Add webhook**;
+5. Input the following details:
+   1. `Title` - enter a descriptive title for this webhook;
+   2. `URL` - enter the value of the `url` key you received after creating the webhook configuration;
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:39]: Maybe rephrase it in a way it will make sense that it comes from Port (the URL)

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+   3. Be sure to keep the "Active" checkbox checked under **Status**;
+   4. Under **Triggers**, select "Push";
+6. Click on **Save** at the bottom of the page.
+
+:::tip
+In order to view the different payloads and events available in BitBucket webhooks, [look here](https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Push)
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:45]: Specify here that the structure of the data can be found here, and it is used by the configuration in the webhook configuration

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  "mappings": [
+    {
+      "blueprint": "bitBucketMicroservice",
+      "filter": ".headers.\"X-Event-Key\" == \"repo:push\"",
+      "entity": {
+        "identifier": "latest_commit",
+        "title": "Latest Commit",
+        "properties": {
+          "commit_hash": ".body.push.changes[0].new.target.hash",
+          "author": ".body.push.changes[0].new.target.author.user.display_name",
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx:16]: Does it works on squash & merge, and also rebase?
Can you validate that please

Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx:16]: Let's name it `commitHash` and `commitAuthor`, as the identifier of the blueprint is camelcased.
And please remember to update the blueprint as well accrodingly

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+sidebar_position: 10
+description: Keep track of latest commits in your catalog
+---
+
+import CommitBlueprint from './resources/bitbucket/\_example_bitbucket_push_blueprint.mdx'
+import CommitWebhookConfig from './resources/bitbucket/\_example_bitbucket_push_webhook_configuration.mdx'
+
+# BitBucket
+
+In this example you are going to create a webhook integration between [BitBucket](https://bitbucket.org) and Port, which will ingest commit entities.
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:11]: This isn't phrase well, please rephrase it in a way that will explain we will add to an existing microservices in port a data about the latest commit pushes to the default branch

Comment (danielsinai) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:11]: In this example, we will be integrating BitBucket and Port to enhance an existing microservice in Port with data regarding the latest commit pushes to the default branch. This integration will involve setting up a webhook to receive notifications from BitBucket whenever a new commit is made to the default branch, allowing Port to ingest and process the commit entities accordingly.

Comment (danielsinai) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:11]: something like that maybe

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+## Prerequisites
+
+Create the following blueprint definition and webhook configuration:
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:15]:
```suggestion
Create the following blueprint definition and webhook configuration in Port:
```

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+   2. `URL` - enter the value of the `url` key you received after creating the webhook configuration;
+   3. Be sure to keep the "Active" checkbox checked under **Status**;
+   4. Under **Triggers**, select "Push";
+6. Click on **Save** at the bottom of the page.
+
+:::tip
+In order to view the different payloads and events available in BitBucket webhooks, [look here](https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Push)
+:::
+
+Done! any change you make to the repository will trigger a webhook event that BitBucket will send to the webhook URL provided by Port. Port will parse the events according to the mapping and update the catalog entities accordingly.
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:48]:
```suggestion
Done! any push to the repository's main branch will trigger a webhook event that will be sent to the webhook URL provided by Port. Port will parse the events according to the mapping and update the catalog entities accordingly.
```

**Review by danielsinai — CHANGES_REQUESTED** [review]
🦢 Great work, some comments

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx (hunk 6 above)
Comment (PeyGis) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx:16]: Reading from the docs, it looks like the[ push event](https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Push) contains a rebased commit.

> Commits that have been force pushed–When you perform a rebase followed by a force push, the commits of the old reference are replaced with all the commits in the new reference. As a result, the array includes the commits part of the new reference.  The array does not include commits that are deleted during a rebase or a force push.

Comment (PeyGis) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx:16]: Also, merge requests are part of a different event [pullrequest:fulfilled](https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Pull-request)

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md (hunk 7 above)
Comment (PeyGis) [reply docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:11]: thanks for this

**Review by danielsinai — APPROVED** [review]
👍

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  "identifier": "bitbucketMapper",
+  "title": "BitBucket Commit Mapper",
+  "description": "A webhook configuration for commit events from BitBucket",
+  "icon": "BitBucket",
+  "mappings": [
+    {
+      "blueprint": "bitBucketMicroservice",
+      "filter": ".headers.\"X-Event-Key\" == \"repo:push\"",
+      "entity": {
+        "identifier": "latest_commit",
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx:12]: Should be the repo name

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+        "title": "Latest Commit",
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx:13]: Should be the repo name

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx [hunk 12]
```diff
@@ -0,0 +1,25 @@
+```json showLineNumbers
+{
+  "identifier": "bitbucketMapper",
+  "title": "BitBucket Commit Mapper",
+  "description": "A webhook configuration for commit events from BitBucket",
+  "icon": "BitBucket",
+  "mappings": [
+    {
+      "blueprint": "bitBucketMicroservice",
+      "filter": ".headers.\"X-Event-Key\" == \"repo:push\"",
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/resources/bitbucket/_example_bitbucket_push_webhook_configuration.mdx:10]: Where the filter is looking for the main branch?

File: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+# BitBucket
+
+In this example, we will be integrating [BitBucket](https://bitbucket.org) and Port to enhance an existing microservice in Port with data regarding the latest commit pushes to the default branch. This integration will involve setting up a webhook to receive notifications from BitBucket whenever a new commit is made to the default branch, allowing Port to ingest and process the commit entities accordingly.
+
+## Prerequisites
+
+Create the following blueprint definition and webhook configuration in Port:
+
+<details>
+<summary>Microservice blueprint</summary>
```
Comment (danielsinai) [inline docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/bitbucket.md:18]: Add a note that explaining that the blueprint microservice can be extended, and that's just an example

**Review by danielsinai — CHANGES_REQUESTED** [review]
Went over it again, seems like some things are yet to be finished
